wpimath/src/main/java/edu/wpi/first/math/trajectory/Trajectory.java

```diff
@@ -92,8 +98,13 @@ public List<State> getStates() {
    *
    * @param timeSeconds The point in time since the beginning of the trajectory to sample.
    * @return The state at that point in time.
+   * @throws IllegalArgumentException if the trajectory has no states.
    */
   public State sample(double timeSeconds) {
+    if (m_states.isEmpty()) {
+      throw new IllegalArgumentException("Trajectory cannot be sampled if it has no states.");
```

Comment:
Wouldn't an IllegalStateException be more fitting? IllegalArgument isn't really appropriate when it's not caused by a parameter immediately passed in.